components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/IconType.kt

```diff
@@ -0,0 +1,25 @@
+package io.github.composegears.valkyrie.parser
+
+import java.nio.file.Path
+import kotlin.io.path.extension
+
+@PublishedApi
+internal enum class IconType(val extension: String) {
+    SVG("svg"),
+    XML("xml"),
+    ;
+
+    companion object {
+        fun from(extension: String): IconType? = when {
+            extension.isSvg -> SVG
+            extension.isXml -> XML
+            else -> null
+        }
+    }
+}
+
+inline val String?.isSvg: Boolean get() = equals(other = IconType.SVG.extension, ignoreCase = true)
```

Comment:
Is it really should be nullable?)

Reply:
Simplify for `extension != null && (extension.isSvg() || extension.isXml())`.